src/services/v2/on-chain/mover/debit-card/DebitCardOnChainService.ts

```diff
@@ -0,0 +1,304 @@
+import { ZeroXAPIService } from '@/services/v2/api/0x';
+import { OnChainServiceError } from '@/services/v2/on-chain';
+import { MoverOnChainService } from '@/services/v2/on-chain/mover';
+import { CompoundEstimateWithUnwrapResponse } from '@/services/v2/on-chain/mover/debit-card/types';
+import { HolyHandContract } from '@/services/v2/on-chain/mover/types';
+import { WrappedTokenDCult } from '@/services/v2/on-chain/wrapped-tokens/dCULT/token';
+import { WrappedTokenGALCX } from '@/services/v2/on-chain/wrapped-tokens/gALCX/token';
+import { IWrappedToken } from '@/services/v2/on-chain/wrapped-tokens/WrappedToken';
+import { WrappedTokenWXBTRFLY } from '@/services/v2/on-chain/wrapped-tokens/wxBTRFLY/token';
+import { WrappedTokenYearn } from '@/services/v2/on-chain/wrapped-tokens/yearn/token';
+import { addSentryBreadcrumb } from '@/services/v2/utils/sentry';
+import { sameAddress } from '@/utils/address';
+import { toWei } from '@/utils/bigmath';
+import { Network } from '@/utils/networkTypes';
+import {
+  getCentralTransferProxyAbi,
+  getEURSAssetData,
+  getSlippage,
+  getUSDCAssetData,
+  lookupAddress
+} from '@/wallet/references/data';
+import ethDefaults from '@/wallet/references/defaults';
+import { addresses as yearnSimpleVaultAddresses } from '@/wallet/references/yearnVaultsData';
+import { SmallToken, SmallTokenInfo } from '@/wallet/types';
+
+import { LoaderStep } from '@/components/forms';
+
+export class DebitCardOnChainService extends MoverOnChainService {
+  protected readonly sentryCategoryPrefix = 'debit-card.on-chain.service';
+  specialTokenHandlers: Array<IWrappedToken> = [];
+  swapService: ZeroXAPIService;
+  protected readonly usdcAssetData: SmallTokenInfo;
+  protected readonly eursAssetData: SmallTokenInfo;
+  protected readonly centralTransferProxyAddress: string;
+  protected readonly centralTransferProxyContract: HolyHandContract | undefined;
+
+  constructor(currentAddress: string, network: Network, web3Client: Web3) {
+    super(currentAddress, network, web3Client);
+    this.swapService = new ZeroXAPIService(currentAddress, network);
+    this.usdcAssetData = getUSDCAssetData(network);
+    this.eursAssetData = getEURSAssetData(network);
+    this.centralTransferProxyAddress = lookupAddress(
+      network,
+      'HOLY_HAND_ADDRESS'
+    );
+    this.centralTransferProxyContract = this.createContract(
+      'HOLY_HAND_ADDRESS',
+      getCentralTransferProxyAbi(network)
+    );
+
+    for (const [n, vaults] of Object.entries(yearnSimpleVaultAddresses)) {
+      for (const vault of vaults) {
+        this.specialTokenHandlers.push(
+          new WrappedTokenYearn(vault.vaultToken.address, n as Network)
+        );
+      }
+    }
+
+    this.specialTokenHandlers.push(new WrappedTokenWXBTRFLY(Network.mainnet));
+    this.specialTokenHandlers.push(new WrappedTokenDCult(Network.mainnet));
+    this.specialTokenHandlers.push(new WrappedTokenGALCX(Network.mainnet));
+  }
+
+  public estimateTopUpCompound = async (
```

Comment:
```suggestion
  public async estimateTopUpCompound(
```
So this will not be a class member but a function